cmd/sops/subcommand/updatekeys/updatekeys.go

```diff
@@ -0,0 +1,198 @@
+package updatekeys
+
+import (
+	"fmt"
+
+	"log"
+
+	"path/filepath"
+
+	"os"
+
+	"strings"
+
+	"github.com/fatih/color"
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/cmd/sops/common"
+	"go.mozilla.org/sops/config"
+	"go.mozilla.org/sops/keys"
+	"go.mozilla.org/sops/keyservice"
+)
+
+type Opts struct {
+	InputPath   string
+	GroupQuorum int
+	KeyServices []keyservice.KeyServiceClient
+	Interactive bool
+	ConfigPath  string
+}
+
+func UpdateKeys(opts Opts) error {
+	path, err := filepath.Abs(opts.InputPath)
+	if err != nil {
+		return err
+	}
+	info, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if info.IsDir() {
+		filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+			if info.IsDir() {
+				return nil
+			}
+			ignoredPaths := []string{
+				"/.git/",
```

Comment:
As a general rule, I don't like hardcoding exceptions. I'm also not sure that's needed because `.git` store binary blobs, not sops file, and `.sops.yaml` doesn't have sops metadata.

Reply:
The reasoning for this was to eliminate hundreds of error messages when running it, but if we go the route of just accepting a single file, it won't be needed at all.